src/vs/workbench/parts/debug/electron-browser/replViewer.ts

```diff
@@ -292,113 +285,130 @@ export class ReplExpressionsRenderer implements IRenderer {
-						// only make sense in combination with text-background ranges which we currently not support
-						let token = document.createElement('span');
-						token.className = '';
-						while (codes.length > 0) {
-							code = codes.pop();
-							let parsedMode = parseInt(code, 10);
-							if (token.className.length > 0) {
-								token.className += ' ';
-							}
-							if ((parsedMode >= 30 && parsedMode <= 37) || (parsedMode >= 90 && parsedMode <= 97)) {
-								token.className += 'code' + parsedMode;
-							} else if (parsedMode === 1) {
-								token.className += 'code-bold';
-							} else if (parsedMode === 4) {
-								token.className += 'code-underline';
-							}
-						}
+				while (currentPos < textLength) {
+					const char: string = text.charAt(currentPos);
+					ansiSequence += char;
 
-						// we need a tokens container now
-						if (!tokensContainer) {
-							tokensContainer = document.createElement('span');
-						}
+					currentPos++;
 
-						// flush text buffer if we have any
-						if (buffer) {
-							this.insert(this.linkDetector.handleLinks(buffer), currentToken || tokensContainer);
-							buffer = '';
-						}
+					if (char.match(/^[ABCDHIJKfhmpsu]$/)) {
+						sequenceFound = true;
+						break;
+					}
 
-						currentToken = token;
+				}
 
-						// get child until deepest nested node is found
-						let childPointer: Node = tokensContainer;
-						while (childPointer.hasChildNodes() && childPointer.firstChild.nodeName !== '#text') {
-							childPointer = childPointer.firstChild;
+				if (sequenceFound) {
+
+					// Flush buffer with previous styles.
+					this.appendStylizedStringToContainer(root, buffer, styleNames);
+					buffer = '';
+
+					/*
+					 * Certain ranges that are matched here do not contain real graphics rendition sequences. For
+					 * the sake of having a simpler expression, they have been included anyway.
+					 */
+					if (ansiSequence.match(/^(?:[39][0-7]|[0-8]|39)(?:;(?:[39][0-7]|[0-8]|39))*;?m$/)) {
+
+						const styleCodes: number[] = ansiSequence.slice(0, -1)	// Remove final 'm' character.
+							.split(';')											// Separate style codes.
+							.filter(elem => elem !== '')						// Filter empty elems as '34;m' -> ['34', ''].
+							.map(elem => parseInt(elem, 10));					// Convert to numbers.
+
+						for (let code of styleCodes) {
+							if (code === 0) {
+								styleNames = [];
+							} else if (code === 1) {
+								styleNames.push('code-bold');
+							} else if (code === 4) {
+								styleNames.push('code-underline');
+							} else if ((code >= 30 && code <= 37) || (code >= 90 && code <= 97)) {
+								styleNames.push('code-fg-' + code);
+							} else if (code === 39) {
+								styleNames = styleNames.filter(style => !style.match(/^code-fg-\d+$/));
+							}
 						}
-						childPointer.appendChild(token);
 
-						i = index;
+					} else if (ansiSequence.match(/^.*[ABCDHIJKfhmpsu]$/)) {
```

Comment:
This regex seems to be duplicated please extract it above with a clear name

Reply:
Now that I think about it, we don't even need this one. I'll remove it completely.

Reply:
Now that I think about it, we don't even need this one. I'll remove it completely.